src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -758,6 +761,48 @@ private void handleViewPinnedMessages() {
     startActivity(intent);
   }
 
+  public void handleEmojiReaction(MessageRecord messageRecord, ReactionsHandler handler){
+
+    isEmojiReactionMode = true;
+
+    onEmojiToggle();
+    container.show(composeText, emojiDrawerStub.get()); //incase toggle doesn't work
+    inputPanel.setVisibility(View.GONE);
+    setEmojiKeyboardListener(messageRecord, handler);
+
+    // Send syncronization message
+    // handleEmojiDisplay(emoji);
+  }
+
+  private void setEmojiKeyboardListener(MessageRecord messageRecord, ReactionsHandler handler) {
+    this.emojiDrawerStub.get().setEmojiEventListener(new EmojiDrawer.EmojiEventListener() {
+      @Override
+      public void onKeyEvent(KeyEvent keyEvent) {
+        Log.w("Daniel", "event: " + keyEvent + "code:" + keyEvent.getKeyCode());
```

Comment:
Is this log priority warning? (TA said we should use the correct logging priorities)